include/CL/sycl/vec.hpp

```diff
@@ -63,7 +63,7 @@ class vec : public detail::small_array<DataType,
 
 
   // Inherit of all the constructors
-  using typename basic_type::small_array;
+  using basic_type::small_array;
```

Comment:
I am not sure we can remove this typename.
It breaks 3 examples with Clang++ 3.9 because here
```
using basic_type::small_array;
```
is actually a shortcut for
```
using detail::small_array<DataType,
                          vec<DataType, NumElements>,
                          NumElements>::small_array;
```
which is a dependent type, so I have the feeling the typename is useful here.